### PR TITLE
Fix web client leaks

### LINKED ISSUE #1530: Web client leaks Backbone views
On the web client front page, simply navigating between the Collections, Users, Group leaks Backbone views. This can be confirmed in Chrome by taking heap snapshots and examining 'Detached DOM tree' entries. [Backbone Debugger](https://github.com/Maluen/Backbone-Debugger) also shows the leaking objects.

Update: Backbone Debugger view list isn't compatible with how we manage views, so it shouldn't be trusted in this case.

### PR DESCRIPTION
**Pull request by msmolens**
Fix various DOM tree leaks in the web client.

Fixes #1530.

### CODE REVIEW

File: clients/web/src/dialog.js [hunk 1]
```diff
@@ -74,7 +74,9 @@ function confirm(params) {
     }, params);
     $('#g-dialog-container').html(ConfirmDialogTemplate({
         params: params
-    })).girderModal(false);
+    })).girderModal(false).one('hidden.bs.modal', function () {
```
Comment (zachmullen) [inline clients/web/src/dialog.js]: Is `.one()` an actual function, or is that a typo?

Comment (zachmullen) [inline clients/web/src/dialog.js]: To answer my own question, this is [a jquery function](http://api.jquery.com/one/), not a backbone function.

Comment (msmolens) [inline clients/web/src/dialog.js]: Right:

> The .one() method is identical to .on(), except that the handler [...] is unbound after its first invocation.

File: clients/web/src/views/body/AssetstoresView.js [hunk 2]
```diff
@@ -61,11 +62,24 @@ var AssetstoresView = View.extend({
...
     },
 
+    destroy: function () {
+        this._destroyPlots();
+        View.prototype.destroy.call(this);
+    },
+
+    _destroyPlots: function () {
+        for (let plot of this.plots) {
+            plot.data('jqplot').destroy();
```
Comment (zachmullen) [inline clients/web/src/views/body/AssetstoresView.js]: So jqplots bind themselves to global scope in some way?

Comment (msmolens) [inline clients/web/src/views/body/AssetstoresView.js]: Without this the views leave detached DOM trees with HTML canvas elements, created by jqplot, at the roots.
https://github.com/jqPlot/jqPlot/blob/460412a84e603a62eea0ecd90057cbaac3e2d9c8/src/jqplot.core.js#L2843

Comment (zachmullen) [inline clients/web/src/views/body/AssetstoresView.js]: Ew, gross. It is a rather dated library.

**Comment by zachmullen** [comment]
@msmolens I had tried investigating this some time ago and discovered one source of leaking was through tooltips, did you notice that as well?

**Comment by msmolens** [comment]
@zachmullen Yes, in https://github.com/girder/girder/pull/1992/commits/b5d1c4c27b3bd740fad984c7a8a947a0c8300c26 it mentions this:

> DOM elements created by Bootstrap tooltips tended to exhibit this behavior

**Comment by zachmullen** [comment]
Great, sorry I missed that message.

**Comment by msmolens** [comment]
@zachmullen Thanks for the review.

**Comment by zachmullen** [comment]
Thanks for the fixes! 🥂

**Pull request merged**